dgm-builder/src/main/java/com/cloudbees/groovy/cps/tool/Translator.java

```diff
@@ -357,26 +351,28 @@ public JExpression visitMethodInvocation(MethodInvocationTree mt, Void __) {
                                 .arg(n(mst.getIdentifier()));
                     }
                 } else
-                if (ms instanceof JCIdent) {
+                if (ms instanceof IdentifierTree) {
```

Comment:
BTW acc. to docs you are never supposed to use `instanceof` with these APIs; rather you check some type enum and then cast. I doubt it matters in practice.

Reply:
Ok, we could check `Tree.getKind()` instead and then cast but IDK what the difference would be. There were maybe 2-3 places with similar code before my changes. Do you have a link to the docs though? I see such a warning for `Element` but not for `Tree`: https://docs.oracle.com/en/java/javase/16/docs/api/java.compiler/javax/lang/model/element/Element.html.

Reply:
Maybe this only applies to `Element` and not `Tree` then.